content/en/platform/corda/5.0/developing-applications/getting-started/utxo-advanced-ledger-extensions/advanced-design.md

```diff
@@ -0,0 +1,334 @@
+---
+date: '2023-06-01'
+title: "Advanced Contract Design"
+project: corda
+version: 'Corda 5.0'
+menu:
+  corda5:
+    identifier: corda5-utxo-ledger-extensions-advanced-design
+    parent: corda5-utxo-advanced-ledger-extensions
+    weight: 4600
+section_menu: corda5
+---
+
+# Advanced Contract Design
+
+All the contract design issues that have been highlighted above are implemented by the Corda 5 Advanced UTXO Extensions library, and are included in all the specific implementations; for example, chainable, fungible and identifiable contracts.
+
+## Base API
+
+Module: base
+
+Package: com.r3.corda.ledger.utxo.base
+
+The base API provides the underlying component model for designing extensible contracts with delegated contract verification constraint logic, as well as other components which allow CorDapp developers to better express intent throughout their applications.
+
+## Chainable API
+
+Module: chainable
+
+Package: com.r3.corda.ledger.utxo.chainable
+
+The chainable API provides the component model for designing chainable states and contracts. Chainable states represent strictly linear state chains, where every state in the chain points to the previous state in the chain. This could be thought of as a similar concept to a blockchain, where each new block points to the previous block.
+
+## Designing a Chainable State
```

Comment:
Shouldn't this heading be a `###`, as it is  sub-chapter of  `## Chainable API`? Everything seems to be pretty much on the same hierarchy level in this doc which makes it hard to parse.

Reply:
Fixed the hierarchy and titles of all headings for consistency.